src/execution/operator/aggregate/physical_ungrouped_aggregate.cpp

```diff
@@ -7,9 +7,151 @@
 #include "duckdb/main/client_context.hpp"
 #include "duckdb/parallel/thread_context.hpp"
 #include "duckdb/planner/expression/bound_aggregate_expression.hpp"
+#include "duckdb/execution/radix_partitioned_hashtable.hpp"
+#include "duckdb/parallel/event.hpp"
+#include "duckdb/common/unordered_set.hpp"
+#include "duckdb/common/algorithm.hpp"
+#include <functional>
 
 namespace duckdb {
 
+DistinctAggregateData::DistinctAggregateData(Allocator &allocator, const vector<unique_ptr<Expression>> &aggregates,
```

Comment:
Can `DistinctAggregateData` be in its own header/implementation file as well? Just like `GroupedAggregateData`.